File: 2020/1.py
values = []

with open("1.txt") as file:
    for line in file:
        values.append(int(line))

stopLooping = False
for i in range(len(values)):
    if stopLooping:
        break
    for value in values:
        if value + values[i] == 2020:
            print(value*values[i])
            stopLooping = True
            break

stopLooping = False
for i in range(len(values)):
    if stopLooping:
        break
    for j in range(len(values)):
        if stopLooping:
            break
        for value in values:
            if value + values[i] + values[j] == 2020:
                print(value*values[i]*values[j])
                stopLooping = True
                break

File: 2020/Lua/1.lua
local nums = {}
for line in io.lines("1.txt") do
  table.insert(nums, tonumber(line))
end

for i = 1, #nums do
  for j = 2, #nums do
    if nums[i] + nums[j] == 2020 then
      print(nums[i] * nums[j])
    end
    for k = 3, # nums do
     if nums[i] + nums[j] + nums[k] == 2020 then
      print(nums[i] * nums[j] * nums[k]) 
     end
    end
  end
end
